src/Item_Devices.php

```diff
@@ -1230,7 +1236,10 @@ public static function addDevicesFromPOST($input)
                         $link->update($update_input);
                     }
                 }
-                if (isset($input['new_devices'])) {
+                if (
+                    isset($input['new_devices'])
+                    && ($link instanceof Item_Devices)
```

Comment:
Maybe not needed anymore with the previous recommendation of not using `getType`.